photonlib-java-examples/simaimandrange/src/main/java/frc/robot/sim/VisionSim.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package frc.robot.sim;
+
+import static frc.robot.Constants.*;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.math.util.Units;
+import java.util.List;
+import org.photonvision.PhotonCamera;
+import org.photonvision.estimation.TargetModel;
+import org.photonvision.simulation.PhotonCameraSim;
+import org.photonvision.simulation.SimCameraProperties;
+import org.photonvision.simulation.VisionSystemSim;
+import org.photonvision.simulation.VisionTargetSim;
+
+/**
+ * This class handles the simulation of the camera and vision target on the field. Updating this
+ * class will update the camera data, reflecting what the simulated camera sees.
+ *
+ * <p>This class and its methods are only relevant during simulation. While on the real robot, the
+ * real camera data is used instead.
+ */
+public class VisionSim {
+    // ----- Simulation specific constants
+    // 2020 High goal target shape
+    // See
+    // https://firstfrc.blob.core.windows.net/frc2020/PlayingField/2020FieldDrawing-SeasonSpecific.pdf
+    // page 208
+    private static final TargetModel kTargetModel =
+            new TargetModel(
+                    List.of(
+                            new Translation3d(0, Units.inchesToMeters(-9.819867), Units.inchesToMeters(-8.5)),
+                            new Translation3d(0, Units.inchesToMeters(9.819867), Units.inchesToMeters(-8.5)),
+                            new Translation3d(0, Units.inchesToMeters(19.625), Units.inchesToMeters(8.5)),
+                            new Translation3d(0, Units.inchesToMeters(-19.625), Units.inchesToMeters(8.5))));
+
+    // Simulated camera properties. These can be set to mimic your actual camera.
```

Comment:
do all of these need to be set or can the user only set some of them? what are some good defaults for these values / how can the user find them

Reply:
These are set to a "perfect" 960x720 90 deg FOV camera without lag or noise by default. There are also static example fields for reference. It will be explained in the docs simulation page/example walkthrough.